src/relocatable.rs

```diff
@@ -56,7 +56,7 @@ impl PyRelocatable {
             }
             PyMaybeRelocatable::RelocatableValue(address) => {
                 if self.segment_index != address.segment_index {
-                    return Err(VirtualMachineError::DiffIndexSub).map_err(to_py_error)?;
+                    return Err(VirtualMachineError::NoSignatureBuiltin).map_err(to_py_error)?;
```

Comment:
This error doesn't make sense here, the error should be `MathError::RelocatableSubDiffIndex`